common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -410,7 +414,19 @@ public boolean endsWith(final UTF8String suffix, int collationId) {
     if (collationId == CollationFactory.LOWERCASE_COLLATION_ID) {
       return this.toLowerCase().endsWith(suffix.toLowerCase());
     }
-    return matchAt(suffix, numBytes - suffix.numBytes, collationId);
+    return collatedEndsWith(suffix, collationId);
+  }
+
+  private boolean collatedEndsWith(final UTF8String suffix, int collationId) {
+    if (suffix.numBytes == 0 || this.numBytes == 0) {
+      return suffix.numBytes==0;
+    }
+    if (suffix.numChars() > this.numChars()) {
+      return false;
+    }
+    return CollationFactory.getStringSearch(
+            this.substring(this.numChars()-suffix.numChars(), this.numChars()),
```

Comment:
identation is a bit weird here. 4 spaces should be more appropriate? Please align with the rest of file.

Reply:
+1

this might be useful: https://github.com/databricks/scala-style-guide

Reply:
> identation is a bit weird here. 4 spaces should be more appropriate? Please align with the rest of file.

Followed [Spacing and Indention rules](https://github.com/databricks/scala-style-guide?tab=readme-ov-file#spacing-and-indentation).